conformance/lang/expressions/error_constructor.balt

```diff
@@ -0,0 +1,526 @@
+
+type E1 any;
+
+type E2 any|error;
+
+function errorFunction() {
+    error err1 = error E1("error message"); // @error cannot create a new error value from 'E1'
+    error err2 = error E2("error message"); // @error cannot create a new error value from 'E2'
+    var err3 = error E1("error message"); // @error cannot create a new error value from 'E1'
+    var err4 = error E2("error message"); // @error cannot create a new error value from 'E2'
+}
+
+Test-Case: error
+Description: Test the effect of specified error-type-reference is same as making the contextually expected type be that specified type reference.
+Labels: error-constructor-expr, record-type, error-type, var, string, module-type-defn
+
+type E error<record {|string message;|}>;
+
+function errorFunction() {
+    E err1 = error("error message"); // @error missing error detail arg for error detail field 'message'
+
+    var err2 = error E("error message"); // @error missing error detail arg for error detail field 'message'
+
+    error err3 = error E("error message"); // @error missing error detail arg for error detail field 'message'
+}
+
+Test-Case: error
+Description: Test when there is no applicable contextually expected type,
+             then it is the same as if there were a contextually expected type of error.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, string, error-type
+
+type E error<record {|string message?;|}>;
+
+function errorFunction() {
+    var err1 = error("error message");
+    int x1 = err1; // @error incompatible types: expected 'int', found 'error'
+
+    var err2 = error("error message", msg = "ERROR!", fatal = false);
+    int x2 = err2; // @error incompatible types: expected 'int', found 'error'
+
+    var err3 = error("error message", error("ERROR!"));
+    int x3 = err3; // @error incompatible types: expected 'int', found 'error'
+
+    var err4 = error("error message", error("ERROR!"), msg = "ERROR!", fatal = false);
+    int x4 = err4; // @error incompatible types: expected 'int', found 'error'
+}
+
+Test-Case: error
+Description: Test the first positional-arg is of type string, and the second positional-arg is of type error?
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, optional-type, string
+
+type E error<record {|string message?;|}>;
+
+function errorFunction() {
+    error err = error("error message");
+
+    error err1 = error(err); // @error incompatible types: expected 'string', found 'error'
+    error err2 = error("error message", "error cause"); // @error incompatible types: expected 'error?', found 'string'
+
+    error err3 = error E(err); // @error incompatible types: expected 'string', found 'error'
+    error err4 = error E("error message", "error cause"); // @error incompatible types: expected 'error?', found 'string'
+}
+
+Test-Case: output
+Description: Test the second positional-arg is by default nil and specified type otherwise.
```

Comment:
Shall we change the description a bit since we have not really tested whether the second positional-arg is of the specified type here?

Reply:
Please check with the new changes. Now the description is matched.